### PR TITLE
PSP-9133 : Implement check/warning on frontend that backend and datab…

### PR DESCRIPTION
**Pull request by eddherrera**
…ase versions are as expected.

image

### CODE REVIEW

File: source/backend/dal/Repositories/SystemConstantRepository.cs [hunk 1]
```diff
@@ -35,6 +35,12 @@ public IEnumerable<PimsStaticVariable> GetAll()
         {
             return this.Context.PimsStaticVariables.AsNoTracking().ToArray();
         }
+
+        public string GetDataBaseVersion()
+        {
+            return this.Context.PimsStaticVariables.Where(x => x.StaticVariableName == "DBVERSION")
+                .AsNoTracking().FirstOrDefault().StaticVariableValue ?? string.Empty;
```
Comment (asanchezr) [inline source/backend/dal/Repositories/SystemConstantRepository.cs:42]: There is potential for a null-pointer exception in this code if the database didn't have an entry with key "DBVERSION" in the static variables table. Granted, we would never see this as we populate that table but better avoid the NPE if possible.

If no DBVERSION value is in the table, then **FirstOrDefault()** returns null and the call to get the **.StaticVariableValue** would throw an exception. We would just need to replace with:

```cs
...AsNoTracking().FirstOrDefault()?.StaticVariable ?? string.Empty
```

File: source/frontend/src/components/common/ApiVersionInfo.tsx [hunk 2]
```diff
@@ -27,11 +32,78 @@ export const ApiVersionInfo = () => {
...
+  const frontEndDBVersion = findDBVersion(frontEndVersion);
+
+  const apiVersionMissmatch = version?.informationalVersion !== frontEndVersion;
+  const dbVersionMissmatch = version?.dbVersion !== frontEndDBVersion;
+
+  const versionMissmatchMsg = (apiMissmatch: boolean, dbMissmatch: boolean): string => {
+    let msg = '';
+    if (apiMissmatch || dbMissmatch) {
+      msg = msg.concat(
+        `Warning: There is a version missmatch with the backend.
```
Comment (asanchezr) [inline source/frontend/src/components/common/ApiVersionInfo.tsx:59]: typo: missmatch -> mismatch

File: source/frontend/src/components/common/ApiVersionInfo.tsx [hunk 3]
```diff
@@ -27,11 +32,78 @@ export const ApiVersionInfo = () => {
...
+  display: flex;
+  gap: 10px;
+  flex-grow: 1;
+  flex-direction: row;
+  justify-content: flex-end;
+  align-items: center;
+`;
+
+export const VersionMissmatchDiv = styled(InlineFlexDiv)`
+  text-transform: uppercase;
```
Comment (asanchezr) [inline source/frontend/src/components/common/ApiVersionInfo.tsx:100]: I don't think this style is needed. We don't want the warning message in all caps

**Review by asanchezr — APPROVED** [review]
See comments

File: source/frontend/src/components/common/ApiVersionInfo.tsx (hunk 3 above)
Comment (eddherrera) [reply source/frontend/src/components/common/ApiVersionInfo.tsx]: removed

File: source/frontend/src/components/common/ApiVersionInfo.tsx (hunk 2 above)
Comment (eddherrera) [reply source/frontend/src/components/common/ApiVersionInfo.tsx]: Updated

**Pull request merged**